### ISSUE TITLE
License conflict

### ISSUE THREAD

**Issue by buzzwick** [microsoft/AL-Go#632]
Deploying from GitHub to a new client BC Sandbox environment (this has now happened for two clients), we get

Deploying to WRDEMO failed.  Status Code 422 : Unprocessable Entity Your license does not grant you the following permissions on TableData Published Application: IndirectInsert for company CRONUS USA, Inc..  To view details about your permissions, see the Effective Permissions page. 

Our AL-Go repository is using a URL to the most current BC v22 .bclicense file. We were able to deploy from AL-Go to a new environment in the UK without this error. Just want to eliminate any source of this error from the AL-Go side.

**Comment by freddydk** [comment]
The license specified in AL-Go for GitHub is ignored when deploying to environments.
I assume that you will get the same problem if you deploy manually to these customers, maybe the user used for the deployment doesn't have sufficient rights?
Are you using Impersonation or S2S?
Which permissions are assigned to the user or app?
Thanks

**Comment by buzzwick** [comment]
That confirms what I wanted to know--no license involvement from the GitHub side. Since creating this issue--in fact, just yesterday--we started getting more informative messages from MS about what was going on. Apparently the admin user we were using to authenticate and create our auth token had been given Global Admin rights to the environment, but in order to deploy extensions they need to be assigned an actual license through MS 365. Working through that now.

As for deploying manually--we tried that, but as our registered Object ID range only applies to AppSource apps, we found manual deployment would not work. Deployment from AL-Go does, however. Why, I would not know.

Thanks for clearing up the license issue.
